### DISCUSSION TITLE
RPM db does not upgrades when upgrading from `4.11.3` to `4.18.2`

### DISCUSSION THREAD

**Discussion by mjcr99** [rpm-software-management/rpm#3431] [Q&A]
**Describe the bug**
Hi team,

I'm working on a Centos 7 Docker container which brings RPM version `4.11.3`. I'm able to cover the required deps and install the `4.18.2` version, but I can not migrate the native rpm db to the installed one and the newer version never knows about the already installed packages.

**To Reproduce**

* Steps to reproduce the behavior:

On a Centos 7 host, install the RPM `4.18.2` with the following command (_having the required deps covered_):

```command
git clone https://github.com/rpm-software-management/rpm.git --branch rpm-4.18.2-release --single-branch && \
    cd rpm && ./autogen.sh && make -j$(nproc) && make install && cd / && rm -rf rpm*
```

Then I try to migrate the installed packages information with the native RPM version as follows:

```command
mkdir -p /usr/local/var/lib/rpm && cp /var/lib/rpm/Packages /usr/local/var/lib/rpm/Packages && \
    /usr/local/bin/rpm --rebuilddb
```

But when querying the installed packages list, it seems to not be updated:
```command
[root@a0607f3a3f16 /]# /usr/local/bin/rpm --version
RPM version 4.18.2
[root@a0607f3a3f16 /]# /usr/local/bin/rpm -qa
```

**Expected behavior**
This procedure has been working for me if the new installed RPM version was the `4.15.1`. With this version the packages list is upgraded and the new version knows the previously installed packages.

Is it expected to not work with the newer mentioned version?

**Comment by mlschroe** [comment]
We removed support for BerkeleyDB in rpm 4.17.0, so your new rpm does not see the old database.

There's still a way to read old databases if you add the `--enable-bdb-ro` option to the `autogen.sh` call. But the new rpm cannot modify the database. The idea is that you use `rpm --rebuilddb` to convert to a supported database format (i.e. sqlite).

**Comment by mjcr99** [comment]
Hi @mlschroe! Thanks for your answer. 

So how could I migrate the old db data to the new rpm version but still be able to modify the migrated data by updating/installing/deleting packages with the the new rpm version?

**Reply by ffesti** [reply]
As soon as you have migrated the database with `rpm --rebuilddb` it is in the new format and can be read by the new rpm version.
